src/check_rules.ml

```diff
@@ -0,0 +1,14 @@
+open Stdune
+
+let exts = [".cmt"; ".cmti"; ".cmi"]
+
+let dev_files p =
+  List.exists ~f:(fun suffix -> Path.is_suffix p ~suffix) exts
```

Comment:
Small nitpick: it seems more direct to me to do `match Path.extension p with | ".cmt" | ... -> true | _ -> false`. It's faster as well as the compiler optimizes pattern matching on strings.

Reply:
Okay, will change it to your suggestion. I actually that my version would have the edge as it wouldn't allocate anything if the closure for `~f` was eliminated.

Reply:
It's true that you skip the allocation and copying